### DISCUSSION TITLE
API for bulk update?

### DISCUSSION THREAD

**Discussion by liaujianjie** [PostgREST/postgrest#1945] [Q&A] [answered]
Presently, there is support for [bulk insertion](https://postgrest.org/en/v8.0/api.html?highlight=bulk#bulk-insert) and [bulk upsertion](https://postgrest.org/en/v8.0/api.html?highlight=bulk#upsert) in the docs. Is there support for bulk updates by primary key?

#### Scenario
- `id` column is the PK

#### HTTP request
```http
PATCH /people HTTP/1.1
Content-Type: application/json

[
  { "id": 1, nationality: "Singaporean" },
  { "id": 2, nationality: "South Korean" }
]
```

#### Expected result

The rows of ids `1` and `2` on the table `people` will have their `nationality` column values updated accordingly to the request body above.

**Comment by steve-chavez** [comment, marked as answer]
Hey @liaujianjie,

Wouldn't UPSERT(`Prefer: resolution=merge-duplicates`) also work for that case? If all the ids already exist, then all the operations would be updates.

**Reply by liaujianjie** [reply]
Upsert requires the inclusion of other properties that I don't actually want to update (which makes sense since it may potentially be an insert). Ideally, I shouldn't have to include the columns I don't wish to update in a batch update operation.

For example, if the `people` table from my example has a 3rd non-null column, I would also need to include that third column when using `PATCH /people` even if there are no changes to the 3rd column.

**Reply by wolfgangwalther** [reply]
Yes, that same requirement came up here: https://github.com/PostgREST/postgrest/issues/1959. We're discussing over there how to implement a proper bulk `PATCH`.
